tests/test_schema_retrieval.py

```diff
@@ -65,24 +67,63 @@ def test_invalid_schema(self):
 
         assert str(e.value) == 'Unknown scheme in "ftp://ftp/is/unsupported": "ftp"'
 
-    @pytest.mark.parametrize('location_type', ('http_url', 'https_url'))
-    def test_insecure_connections_fail(self, location_type, untrust_self_signed_cert, request):
+
+class TestSecureSchemaRetrieval:
+    "Common tests that mainly concern securely retrieving schema files."
+
+    @pytest.fixture(params=('http_url', 'https_url'))
+    def remote_dataset_schema_url(self, request) -> str:
+        "http/https url to datasets.yaml."
+        return str(request.getfixturevalue('schema_file_' + request.param)) + '/datasets.yaml'
+
+    @pytest.fixture(params=('absolute_dir', 'relative_dir', 'file_url', 'https_url'))
+    def dataset_schema_url_or_path(self, request) -> str:
+        "Every kind of url or path to datasets.yaml."
```

Comment:
```suggestion
        "https url and every kind of path to datasets.yaml."
```
Since `http_url` isn't being included here

Reply:
Maybe also consider renaming `dataset_schema_url_or_path` to `dataset_schema_https_url_or_path`? Unless this is getting too messy and not worth it

Reply:
I changed the comment to

> Every kind of URL or path (other than http URL) to datasets.yaml.

`file://` is a kind of URL